File: 3622CheckDevisibility.java


class CheckDivisibility {

    static public boolean checkDivisibility(int n) {
        int sum = 0;
        int product = 1;
        int value = n;

        while (value > 0) {
            int digit = value % 10;
            sum += digit;
            product *= digit;
            value /= 10;
        }

        return n % (sum + product) == 0;
    }
    public static void main(String[] args) {
        int n = 99;
        System.err.println(checkDivisibility(n));
    }
}

File: 3622CheckDevisibility.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        original = n
        digit_sum = 0
        digit_product = 1

        while n > 0:
            digit = n % 10
            digit_sum += digit
            digit_product *= digit
            n = n // 10

        total = digit_product + digit_sum
        return original % total == 0


sol = Solution()
print(sol.checkDivisibility(99))
